Update .NET 10 overview page for Preview 6 (#47428)
dotnet/docs: docs/core/whats-new/dotnet-10/libraries.md

title: What's new in .NET libraries for .NET 10
description: Learn about the updates to the .NET libraries for .NET 10.
titleSuffix: ""
ms.date: 07/16/2025
ms.topic: whats-new
ai-usage: ai-assisted
---

# What's new in .NET libraries for .NET 10

> This article describes new features in the .NET libraries for .NET 10. It has been updated for Preview 6.

Reviewer: grammar. In general, prefer contractions:

```suggestion
This article describes new features in the .NET libraries for .NET 10. It's been updated for Preview 6.
```

These algorithms all continue with the pattern of having a static `IsSupported` property to indicate if the algorithm is supported on the current system.

> .NET 10 includes Windows CNG support for post-quantum cryptography, making these algorithms available on Windows systems with PQC support. For example:

Reviewer: Spell out acronyms on first use:  CNG is "Cryptography API: Next Generation" and PQC is "Post-Quantum Cryptography".

```csharp
using System;
using System.IO;
using System.Security.Cryptography;

private static bool ValidateMLDsaSignature(ReadOnlySpan<byte> data, ReadOnlySpan<byte> signature, string publicKeyPath)
{
    string publicKeyPem = File.ReadAllText(publicKeyPath);

>     using (MLDsa key = MLDsa.ImportFromPem(publicKeyPem))
>     {
>         return key.VerifyData(data, signature);
>     }

Reviewer: Simplify this `using` statement:

```suggestion
    using MLDsa key = MLDsa.ImportFromPem(publicKeyPem);
    return key.VerifyData(data, signature);
```
